test/models/config_test.rb

```diff
@@ -19,6 +44,26 @@ class ConfigTest < ActiveSupport::TestCase
       refute @bad_config.valid?
       assert @bad_config.errors.messages[:config_key].include? "can't be blank"
     end
+
+    it 'should validate URLs' do
+      c = Config.find_or_create_by(config_key: 'fax_service')
+      
+      # good
+      c.config_value = { options: ["https://good.com"] }
+      assert c.valid?
+
+      # good but needs cleanup
+      c.config_value = { options: ["www.efax.com/path"] }
```

Comment:
Hm, can you say more about why this is allowed still? I would think that something that would get changed by cleanup should be invalid, but interested in your perspective here.

Reply:
The cleanup is run by `before_validation` - perhaps I should make this more clear in the test. The cleanup is run automatically when we call `c.valid?` and adds `https://`, making it a valid URL. So, for example, after L57 is run, the URL is now `https://www.efax.com/path`.